components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -2776,6 +2784,109 @@ impl ToAnimatedZero for IntermediateSVGPaintKind {
     }
 }
 
+impl Into<NumberOrPercentage> for NonNegativeLengthOrPercentage {
+    fn into(self) -> NumberOrPercentage {
+        self.0.into()
+    }
+}
+
+impl Into<NumberOrPercentage> for NonNegativeNumber {
+    fn into(self) -> NumberOrPercentage {
+        self.0.into()
+    }
+}
+
+impl Into<NumberOrPercentage> for LengthOrPercentage {
+    fn into(self) -> NumberOrPercentage {
+        match self {
+            LengthOrPercentage::Length(len) => NumberOrPercentage::Number(len.to_f32_px()),
+            LengthOrPercentage::Percentage(p) => NumberOrPercentage::Percentage(p),
+            LengthOrPercentage::Calc(_) => {
+                // TODO: We need to treat calc value.
+                // https://bugzilla.mozilla.org/show_bug.cgi?id=1386967
```

Comment:
I wonder if we should reference the Gecko bug here. I don't think we ever expect calc() values here? If we do get them then perhaps it means we are failing to reject them earlier? Perhaps failing to fall back to discrete interpolation?

I wonder if this should be a debug assert instead.